src/sliceSelector.ts

```diff
@@ -0,0 +1,30 @@
+export type Selector<S, R> = (state: S) => R
+
+export function createSliceSelector<S = any>(): Selector<S, S>
+export function createSliceSelector<
+  S extends { [key: string]: any } = any,
+  R = any
+>(slice: string): Selector<S, R>
```

Comment:
what is `R`?

Reply:
I'm guessing it's the type of the return value of the selector